app/post_upgrade.go

```diff
@@ -104,13 +105,22 @@ func (u *postUpgrader) waitManagerUpgradeComplete() error {
 			continue
 		}
 		complete = true
+	PodLookupLoop:
 		for _, pod := range podList.Items {
 			if app, ok := pod.Labels["app"]; !ok || app != types.LonghornManagerDaemonSetName {
 				continue
 			}
-			if len(pod.Spec.Containers) != 1 || pod.Spec.Containers[0].Image != ds.Spec.Template.Spec.Containers[0].Image {
-				complete = false
-				break
+			newImage := ds.Spec.Template.Spec.Containers[0].Image
+			if len(pod.Spec.Containers) != 1 || pod.Spec.Containers[0].Image != newImage {
+				for _, container := range pod.Spec.Containers {
+					if !strings.HasPrefix(container.Name, "longhorn") {
+						continue
+					}
+					if container.Image != newImage {
+						complete = false
+						break PodLookupLoop
+					}
+				}
```

Comment:
Consider adding comments to describe the reason for this condition check, and including an issue link would provide better context and traceability.

Reply:
Added a comment to describe the reason and an issue link. Thanks